pipeline/matching/evaluation/analytics_test.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+
+from pipeline.matching.core.engine import MatchingEngine
+from pipeline.matching.evaluation.analytics import CollectInternalAnalytics
+from pipeline.types import Match, MatchingInput, MatchingInternalAnalytics, User
+
+
+def test_analytics_logic():
+
+    users = [
+        User(uid="A", displayName="User A"),
+        User(uid="B", displayName="User B"),
+        User(uid="C", displayName="User C"),
+        User(uid="D", displayName="User D"),
+        User(uid="E", displayName="User E"),
+    ]
+    matches = [Match(users={"A", "B"}), Match(users={"C", "D"})]
+    user_analytics = CollectInternalAnalytics()
+    actual = user_analytics.n_proposed_matches_per_user(users, iter(matches))
+    expected = MatchingInternalAnalytics(
+        n_proposed_matches_per_user={"A": 1, "B": 1, "C": 1, "D": 1, "E": 0}
+    )
+
+    assert actual == expected
+
+    # This test varifies the behavior of the internal analyzer w/ a simple case
```

Comment:
Typo "verifies", also it is easier for other readers if the comment is at the top of the test:

```python
def test_analytics_logic():
    # This test verifies the behavior of the internal analyzer w/ a simple case
```